File: Android/App/app/src/main/java/com/efalco777/crud_products/presentation/product/model/PresentationProductDetails.kt
package com.efalco777.crud_products.presentation.product.model

data class PresentationProductDetails(
    val id: Int,
    val title: String,
    val description: String,
    val category: String,
    val price: Double,
) {
    companion object {
        // Factory function for creating an empty or default instance
        fun empty() = PresentationProductDetails(
            id = 0,
            title = "",
            description = "",
            category = "",
            price = 0.0
        )
    }
}

File: Flutter/App/lib/presentation/product/model/presentation_product_details.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'presentation_product_details.freezed.dart';

@freezed
class PresentationProductDetails with _$PresentationProductDetails {
  const factory PresentationProductDetails({
    required int id,
    required String title,
    required String description,
    required double price,
    required String category,
    required List<String> images,
  }) = _PresentationProductDetails;

  static const empty = PresentationProductDetails(
    id: 0,
    title: '',
    description: '',
    price: 0,
    category: '',
    images: [],
  );
}
